src/pages/Admin/Venue/Rooms/RoomsForm.tsx

```diff
@@ -52,30 +54,31 @@ export const RoomsForm: React.FC = () => {
     ? parseInt(queryRoomIndexString)
     : undefined;
 
-  useEffect(() => {
-    if (!venueId) return history.replace("/admin");
+  const [{ loading: isLoading }, fetchVenueFromAPI] = useAsyncFn(async () => {
+    const venueSnapshot = await firestore
+      .collection("venues")
+      .doc(venueId)
+      .get();
 
-    const fetchVenueFromAPI = async () => {
-      const venueSnapshot = await firestore
-        .collection("venues")
-        .doc(venueId)
-        .get();
+    if (!venueSnapshot.exists) return history.replace("/admin");
 
-      if (!venueSnapshot.exists) return history.replace("/admin");
+    const data = venueSnapshot.data() as AnyVenue;
+    //find the template
+    const template = ALL_VENUE_TEMPLATES.find(
+      (template) => data.template === template.template
+    );
 
-      const data = venueSnapshot.data() as AnyVenue;
-      //find the template
-      const template = ALL_VENUE_TEMPLATES.find(
-        (template) => data.template === template.template
-      );
+    if (!template || !HAS_ROOMS_TEMPLATES.includes(template.template)) {
+      history.replace("/admin");
+    }
+    setVenue(data as PartyMapVenue);
+  }, [firestore, history, venueId]);
+
+  useEffect(() => {
```

Comment:
You can use `useAsync` and get rid of this `useEffect`

Reply:
@denisdimitrov fixed in 8046f962fca8b50460576d6e7fad9013ae583373